flexmeasures/data/models/data_sources.py

```diff
@@ -10,14 +10,89 @@
 from flask import current_app
 import hashlib
 
+from marshmallow import Schema
+
 
 if TYPE_CHECKING:
     from flexmeasures.data.models.user import User
 
 
-class DataGeneratorMixin:
+class DataGenerator:
+    __data_generator_base__: str | None = None
     _data_source: DataSource | None = None
 
+    _config: dict = None
+
+    _input_schema: Schema | None = None
+    _config_schema: Schema | None = None
+
+    def __init__(self, config: dict | None = None, **kwargs) -> None:
+        """Base class for the Schedulers, Reporters and Forecasters.
+
+        The configuration `config` stores static parameters, parameters that, if
+        changed, trigger the creation of a new DataSource.  Dynamic parameters, such as
+        the start date, can go into the `inputs`. See docstring of the method `DataGenerator.compute` for
+        more details.
+
+
+        Create a new DataGenerator with a certain configuration. There are two alternatives
+        to define the parameters:
+
+            1.  Serialized through the keyword argument `config`.
+            2.  Deserialized, passing each parameter as keyword arguments.
+
+        The configuration is validated using the schema `_config_schema`, to be defined.
+
+        Example:
+
+            The configuration requires the parameters start and end, two datetimes.
+
+            Option 1:
+            dg = DataGenerator(config = {
+                "start" : "2023-01-01T00:00:00+00:00",
+                "end" : "2023-01-02T00:00:00+00:00"
```

Comment:
Not the best example, given that start and end would typically go into the inputs, right?